docs/csharp/nullable-migration-strategies.md

```diff
@@ -86,3 +86,120 @@ This doesn't mean you can't use a nullable type (either value type or reference
+
+```csharp
+class Student
+{
+    [Required]
+    public string FirstName { get; set; }
+
+    [Required]
+    public string LastName { get; set; }
+
+    public string VehicleRegistration { get; set; }
+}
+```
+
+The design intent (indicated in this case by the ``Required`` attribute) suggests that in this system, the ``FirstName`` and ``LastName`` properties are **mandatory**, and therefore not null.
+
+The ``VehicleRegistration`` property is **not mandatory**, so may be null.
+
+When we enable nullable reference types, we want to indicate on our DTO which of the properties may be nullable, consistent with our original intent:
+
+```csharp
+class Student
+{
+    [Required]
+    public string FirstName { get; set; }
+
+    [Required]
+    public string LastName { get; set; }
+
+    public string? VehicleRegistration { get; set; }
+}
+```
+
+For this DTO, the only property that may be null is ``VehicleRegistration``.
+
+However, the compiler raises ``CS8618`` warnings for both ``FirstName`` and ``LastName``, indicating the non-nullable properties are uninitialized.
+
+There are three options available to you that resolve the compiler warnings, in a way that maintains the original intent. Any of these options are valid; you should choose the one that best suits your coding style and design requirements.
+
+### Initialize in the constructor
+
+The ideal way to resolve the uninitialized warnings is to initialize the properties in the constructor:
+
+```csharp
+class Student
+{
+    public Student(string firstName, string lastName)
+    {
+        FirstName = firstName;
+        LastName = lastName;
+    }
+
+    [Required]
+    public string FirstName { get; set; }
+
+    [Required]
+    public string LastName { get; set; }
+
+    public string? VehicleRegistration { get; set; }
+}
+```
+
+This approach only works if the library that you use to instantiate the class supports passing parameters in the constructor.
+
+In addition, a library may support passing *some* properties in the constructor, but not all.
+For example, EF Core supports [constructor binding](/ef/core/modeling/constructors) for normal column properties, but not navigation properties.
+
+Check the documentation on the library that instantiates your class, to understand the extent to which it supports constructor binding.
+
+### Property with nullable backing field
+
+If constructor binding won't work for you, one way to deal with this problem is to have a non-nullable property with a nullable backing field:
+
+```csharp
+private string? _firstName;
+
+[Required]
+public string FirstName
+{
+    set => _firstName = value;
+    get => _firstName
+           ?? throw new InvalidOperationException("Uninitialized " + nameof(FirstName))
+}
+```
+
+In this scenario, if the ``FirstName`` property is accessed before it has been initialized, then we throw an ``InvalidOperationException``, because the API contract has been used incorrectly.
```

Comment:
single back-tick:

```suggestion
In this scenario, if the `FirstName` property is accessed before it has been initialized, then we throw an `InvalidOperationException`, because the API contract has been used incorrectly.
```